sample/src/main/java/com/github/amlcurran/showcaseview/sample/EventsActivity.java

```diff
@@ -0,0 +1,150 @@
+package com.github.amlcurran.showcaseview.sample;
+
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.view.ViewCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.text.SpannableStringBuilder;
+import android.view.MotionEvent;
+import android.view.animation.Interpolator;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.github.amlcurran.showcaseview.OnShowcaseEventListener;
+import com.github.amlcurran.showcaseview.ShowcaseView;
+import com.github.amlcurran.showcaseview.SimpleShowcaseEventListener;
+import com.github.amlcurran.showcaseview.targets.ViewTarget;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class EventsActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_events);
+
+        TextView eventLog = (TextView) findViewById(R.id.events_log);
+        Button customButton = (Button) getLayoutInflater().inflate(R.layout.view_custom_button, null);
+
+        new ShowcaseView.Builder(this)
+                .withMaterialShowcase()
+                .setStyle(R.style.CustomShowcaseTheme3)
+                .setTarget(new ViewTarget(R.id.imageView, this))
+                .setContentTitle("Events")
+                .setContentText("Listening to ShowcaseView events is easy!")
+                .setShowcaseEventListener(new MultiEventListener(new LogToTextListener(eventLog), new ShakeButtonListener(customButton)))
```

Comment:
Maybe extract the event listener in a dedicated local variable for better readability